Octokit/Models/Response/StringEnum.cs

```diff
@@ -0,0 +1,156 @@
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using Octokit.Internal;
+
+namespace Octokit
+{
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    [SuppressMessage("Microsoft.Naming", "CA1711:IdentifiersShouldNotHaveIncorrectSuffix")]
+    public struct StringEnum<TEnum> : IEquatable<StringEnum<TEnum>>
+        where TEnum : struct
+    {
+        private readonly string _stringValue;
+
+        private TEnum? _parsedValue;
+
+        public StringEnum(TEnum parsedValue)
+        {
+            if (!Enum.IsDefined(typeof(TEnum), parsedValue))
+            {
+                throw GetArgumentException(parsedValue.ToString());
+            }
+
+            // Use the SimpleJsonSerializer to serialize the TEnum into the correct string according to the GitHub Api strategy
+            _stringValue = new SimpleJsonSerializer().SerializeEnum(parsedValue as Enum);
```

Comment:
Yep I confirmed this was a bug, and added a failing test to repro it.  If the same `SimpleJsonSerializer` instance tried to deserialize a subsequent enum value that required the Parameter attribute (ie `EncodingType` "ascii" followed by "utf-8") it would throw an exception as it hadn't cached the other enum members, and tried to do a simple `Enum.Parse()` on "utf-8".

I've made the fix identified by @mderriey and confirmed this test now passes, plus changed the `StringEnum` to use a static `SimpleJsonSerializer` and verified the other tests are passing.

Ill push the commits up if you want to have a look @khellang and @mderriey